gpii/node_modules/deviceReporter/src/DeviceReporter.js

```diff
@@ -126,29 +126,28 @@ fluid.defaults("gpii.deviceReporter.live", {
     invokers: {
         get: {
             funcName: "gpii.deviceReporter.live.get",
-            args: ["{arguments}.0", "{deviceReporter}", "{solutionsRegistryDataSource}"]
+            args: ["{arguments}.0", "{deviceReporter}", "{flowManager}.options.fullSolutionsRegistry"]
         }
     }
 });
 
-gpii.deviceReporter.live.get = function (request, deviceReporter, solutionsRegistryDataSource) {
+gpii.deviceReporter.live.get = function (request, deviceReporter, fullSolutionsRegistry) {
     var installedSolutions = [];
-    solutionsRegistryDataSource.get({os: deviceReporter.platformReporter.reportPlatform().id}, function onSuccess(entries) {
-        fluid.each(entries, function (entry, entryId) {
-            if (!installedSolutions.some(function (s) { return s.id === entryId; })) {
-                var foundEntryId = fluid.find(entry.isInstalled, function (installedSolutionsReporter) {
-                    var resolvedName = deviceReporter.nameResolver.resolveName(installedSolutionsReporter.type, "deviceReporter");
-                    if (fluid.invokeGradedFunction(resolvedName, installedSolutionsReporter)) {
-                        return entryId;
-                    }
-                }, null);
-                if  (foundEntryId !== null) {
-                    installedSolutions.push({ "id": foundEntryId });
+    var solutionRegistryEntries = fullSolutionsRegistry[deviceReporter.platformReporter.reportPlatform().id];
+    fluid.each(solutionRegistryEntries, function (entry, entryId) {
```

Comment:
Given we've modified this, it's a good opportunity to pull all of this code out of the callback into a suitable publically named function when we revert it